lldb/tools/lldb-dap/lldb-dap.cpp

```diff
@@ -1364,8 +1364,20 @@ void request_evaluate(const llvm::json::Object &request) {
   std::string expression = GetString(arguments, "expression").str();
   llvm::StringRef context = GetString(arguments, "context");
 
-  if (context == "repl" && g_dap.DetectExpressionContext(frame, expression) ==
-                               ExpressionContext::Command) {
+  if (context == "repl" &&
+      ((!expression.empty() &&
+       g_dap.DetectExpressionContext(frame, expression) ==
+       ExpressionContext::Command) ||
+       (expression.empty() &&
+        g_dap.last_expression_context == ExpressionContext::Command))) {
+    // If the current expression is empty, and the last expression context was
+    // for a  command, pass the empty expression along to the
+    // CommandInterpreter, to repeat the previous command. Also set the
+    // expression context properly for the next (possibly empty) expression.
+    g_dap.last_expression_context = ExpressionContext::Command;
+    // Since the current expression context is not for a variable, clear the
+    // last_nonempty_var_expression field.
+    g_dap.last_nonempty_var_expression.clear();
```

Comment:
It would be cleaner if this gets stored in a sort of discriminated union:

LastExpressionInfo:
  - Case Command (no metadata)  
  - Case Var (expression)

this way it's easier to maintain these two variables correctly in sync

Reply:
I don't think a union will work because last_expression_context is needed and used in all empty expression cases, while for the variable case we ALSO need the last_nonempty_var_expression.

Reply:
Actually I have an idea for a way to make this work with only one variable; let me try that.

Reply:
The idea worked; I've updated the code now.